config/routes.rb

```diff
@@ -3,14 +3,19 @@
   #   resources :posts, only: [:index, :show]
   # end
 
-  get '/', to: 'users#index'
-  get 'users/:id', to: 'users#show'
-  get 'users/:id/posts', to: 'posts#index'
-  get 'users/:id/posts/:id', to: 'posts#show'
+  # get '/users', to: 'users#index'
+  # get 'users/:id' => 'users#show'
+  # get 'users/:id/posts', to: 'posts#index'
+  # get 'users/:id/posts/new', to: 'posts#new', as: 'new_post'
+  # get 'users/:id/posts/:id', to: 'posts#show'
+  post 'users/:user_id/posts/:id/create_like', to: 'likes#create', as: 'like_comment'
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :new, :create, :show] do
+      resources :comments, only: [:new, :create] do
+        resources :likes, only: [:create]
+      end
+    end
```

Comment:
- [ ] Currently when I am trying to like the comment, it is giving me a route error, it's because you have used `likes` route under the `comment` route. 

![Screenshot from 2022-01-22 12-21-07](https://user-images.githubusercontent.com/22434660/150628672-9759e2ab-49e1-4f5e-8ea9-747ab10ec57d.png)

After running rails routes, you have two different routes, Please fix this.

![Screenshot from 2022-01-22 12-33-54](https://user-images.githubusercontent.com/22434660/150629027-32039782-edae-4185-b042-85ada0630a6e.png)



What I recommend is to use a route like this.

```ruby
resources :users, only: [:index, :show] do
  resources :posts, only: [:index, :new, :create]
    resources :posts, only: [:show] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
```